### PR TITLE
Issue 1411 part2

### LINKED ISSUE #1411: Add film + rights tunnel - 6 (technical information)
https://projects.invisionapp.com/d/main#/console/19020838/398261747/preview
ref blockframes/storymapping#14
On Zeplin :
https://app.zeplin.io/project/5ca741b3ed21e79faa266664/screen/5e0b36dcd30a77171ff94f2b

### location
libs/movie/form/version-info (existing component)
libs/movie/display/version-info (existing component)
### name
MovieFormVersionInfoComponent
MovieDisplayVersionInfoComponent

@2amarkina 
image
what formats can user select?
will be format quality changed depends on format?

# E2E Tests to implement 

## Page 6
- [ ] Selects "Dolby SR" for "Sound Format"
- [ ] Selects "SD" for "Format Quality"
- [ ] Selects "Color" for "Color"
- [ ] Toggles "Yes" for "Original language"
- [ ] Selects "English" for "Language" 
- [ ] Clicks "Subtitled" for "English"
- [ ] Clicks on "Next" button
- [ ] Redirects to movie tunnel page 7 Keywords

### PR DESCRIPTION
**Pull request by Maxstgt**
finished page 6
close #1411

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/movie/search/filter.util.ts [hunk 1]
```diff
@@ -23,15 +23,15 @@ function hasLanguage(
   let original = true;
   let dubbed = true;
   let subtitle = true;
-  if (language.original) {
-    original = movie.main.originalLanguages.includes(language.name.toLowerCase() as any);
+/*   if (language.original) {
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/movie/search/filter.util.ts]: Remove this code if you don't need it

File: libs/movie/src/lib/movie/+state/movie.firestore.ts [hunk 2]
```diff
@@ -245,7 +245,7 @@ interface MovieRaw<D> {
   promotionalDescription: MoviePromotionalDescription;
   salesCast: MovieSalesCast;
   salesInfo: MovieSalesInfoRaw<D>;
-  versionInfo: MovieVersionInfo;
+  versionInfo: Partial<{ [language in LanguagesSlug]: MovieLanguageSpecification }>;
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/+state/movie.firestore.ts]: Be sure to tell @phpgeekfr about this changeof data model

File: libs/movie/src/lib/movie/components/import/view-extracted-elements/view-extracted-elements.component.ts [hunk 3]
```diff
@@ -495,7 +494,7 @@ export class ViewExtractedElementsComponent {
         }
 
         // DUBS (Available dubbing(s))
-        if (spreadSheetRow[SpreadSheetMovie.dubbings]) {
+/*         if (spreadSheetRow[SpreadSheetMovie.dubbings]) {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/components/import/view-extracted-elements/view-extracted-elements.component.ts]: Why did you change that ? It's not your code, and I don't really see how it can create an issue as you didn't update the SpreadSheetMovie model.

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 4]
```diff
@@ -1,38 +1,139 @@
 <form [formGroup]="form" fxLayout="column" fxLayoutAlign="center start">
-  <div fxLayout="row" fxLayoutAlign="space-between center" class="full-width">
-    <span class="mat-body">Original language</span>
-    <mat-slide-toggle></mat-slide-toggle>
+  <span class="mat-body">Original Language(s)</span>
+  <div
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: I prefer when you separate into two lines (one for the loop, one for the layout) : 
```typescript
<ng-container *ngFor="let ...">
  <div fxLayout=""..>
```

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 5]
```diff
@@ -1,38 +1,139 @@
...
-      <mat-option>
-        <ngx-mat-select-search
-          [formControl]="languagesFilterCtrl"
-          placeholderLabel="Search..."
-          noEntriesFoundLabel="no matching languages found"
-        ></ngx-mat-select-search>
-      </mat-option>
-      <mat-option *ngFor="let language of languages$ | async" [value]="language.slug">
+  <div>
+    <mat-form-field appearance="outline">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: Why you don't reuse the component I created `movie-form-languages` ?

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 6]
```diff
@@ -1,38 +1,139 @@
...
-      <mat-icon>add</mat-icon>
-    </button>
-    <span class="caption">Add an other Language</span>
-  </div>
+  <form [formGroupName]="language" *ngFor="let language of selectedLanguages$ | async">
+    <ng-container *ngFor="let control of controls.language.controls; let i = index">
+      <ng-container *ngIf="isNotOriginal(language)">
+        <div fxLayout="column">
+          <div fxLayout="row" fxLayoutAlign="center center" fxLayoutGap="20px">
+            <mat-form-field appearance="outline">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: Reuse the movie-form-language component instead of recreating it.

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 7]
```diff
@@ -1,38 +1,139 @@
...
+              <mat-label>Language</mat-label>
+              <mat-select [formControl]="control">
+                <mat-option [value]="language.slug" *ngFor="let language of staticLanguages">
+                  {{ language.label }}
+                </mat-option>
+              </mat-select>
+            </mat-form-field>
+            <div fxLayout="column">
+              <mat-button-toggle-group multiple>
+                <mat-button-toggle
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: You should reuse that part too. You can use `ng-template` for that.

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 8]
```diff
@@ -1,38 +1,139 @@
...
+              <mat-icon>delete</mat-icon>
+            </button>
+            <span class="caption">Delete Additional Language</span>
+          </div>
+        </div>
+      </ng-container>
+    </ng-container>
+  </form>
+
+  <mat-form-field appearance="outline">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: Reuse component

File: libs/movie/src/lib/movie/components/import/view-extracted-elements/view-extracted-elements.component.ts (hunk 3 above)
Comment (Maxstgt) [reply libs/movie/src/lib/movie/components/import/view-extracted-elements/view-extracted-elements.component.ts]: old movie model, breaks app

File: libs/movie/src/lib/movie/+state/movie.firestore.ts (hunk 2 above)
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/+state/movie.firestore.ts]: I'm ok with this `Partial<{ [language in LanguagesSlug]: MovieLanguageSpecification }>;` But it should be `movie.versionInfo.languages =Partial<{ [language in LanguagesSlug]: MovieLanguageSpecification }>`  (draw.io).
@Maxstgt we already had a talk about this last week if I remember well (In the future we could add wind, temperature^^). More seriously, we could have `movie.versionInfo.censored = {some, data} for example`

Comment (phpgeekfr) [reply libs/movie/src/lib/movie/+state/movie.firestore.ts]: @GrandSchtroumpf What do you think about this ?

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html (hunk 6 above)
Comment (Maxstgt) [reply libs/movie/src/lib/movie/form/version-info/version-info.component.html]: no, doesnt come form movie main here

File: libs/movie/src/lib/movie/+state/movie.firestore.ts (hunk 2 above)
Comment (GrandSchtroumpf) [reply libs/movie/src/lib/movie/+state/movie.firestore.ts]: It's fine for me.
@Maxstgt Can you update?

File: libs/movie/src/lib/movie/display/version-info/version-info.component.ts [hunk 9]
```diff
@@ -10,10 +9,11 @@ import { getLabelByCode } from '@blockframes/utils/static-model/staticModels';
 })
 export class MovieDisplayVersionInfoComponent {
 
-  public data: MovieVersionInfo;
-  public getLabelByCode = getLabelByCode;
-  @Input() set info(versionInfo: Partial<MovieVersionInfo>) {
+  public data: MovieLanguageSpecification;
+/*   public getLabelByCode = getLabelByCode;
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/display/version-info/version-info.component.ts]: Add todo / reference to issue

Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/display/version-info/version-info.component.ts]: Add todo / reference to issue

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 10]
```diff
@@ -1,38 +1,81 @@
...
-          noEntriesFoundLabel="no matching languages found"
-        ></ngx-mat-select-search>
-      </mat-option>
-      <mat-option *ngFor="let language of languages$ | async" [value]="language.slug">
-        {{ language.label }}
-      </mat-option>
-    </mat-select>
-  </mat-form-field>
+<span class="mat-body">Original Language(s)</span>
+<div *ngFor="let control of originalLanguages.controls; let i = index" fxLayout="row" fxLayoutAlign="center center"
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: separate logic between 
- *ngFor -> ng-container
- layout -> div

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 11]
```diff
@@ -1,38 +1,81 @@
...
+  fxLayoutGap="20px">
+  <ng-container *ngIf="control.value; else template">
+    <div fxLayout="column" fxLayoutAlign="start center" fxLayoutGap="20px">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: I don't understand why you need this line

File: libs/movie/src/lib/movie/form/version-info/version-info.component.html [hunk 12]
```diff
@@ -1,38 +1,81 @@
...
+        <mat-icon>delete</mat-icon>
+      </button>
+      <span class="caption">Delete Additional Language</span>
+    </div>
   </div>
-</form>
+</ng-container>
+
+<mat-form-field appearance="outline">
+  <mat-label>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.html]: format on 1line

File: libs/movie/src/lib/movie/form/version-info/version-info.component.ts [hunk 13]
```diff
@@ -1,38 +1,151 @@
...
+  selector: '[form] [originalLanguages] movie-form-version-info',
   templateUrl: './version-info.component.html',
   styleUrls: ['./version-info.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class MovieFormVersionInfoComponent implements OnInit {
-
+export class MovieFormVersionInfoComponent implements OnInit, OnDestroy {
   @Input() form: MovieVersionInfoForm;
+  @Input() originalLanguages: FormList<string, FormControl>;
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.ts]: type it like that : `FormList<LanguageSlug, FormStaticField<'LANGUAGE'>>`
Like that you can have strong type and avoid "as LanguageSlug"

File: libs/movie/src/lib/movie/form/version-info/version-info.component.ts [hunk 14]
```diff
@@ -1,38 +1,151 @@
...
+    });
+
+    this.languages$ = this.languageFilterCtrl.valueChanges.pipe(
+      startWith(this.languageFilterCtrl.value),
+      map(language => {
+        return language ? this.languageFilter(language) : this.staticLanguages.slice();
+      })
     );
+
+    this.form.valueChanges.subscribe(console.log);
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/version-info/version-info.component.ts]: Remove this.
